packages/rtk-query-graphql-request-base-query/src/GraphqlBaseQueryTypes.ts

```diff
@@ -27,7 +27,7 @@ export type GraphqlRequestBaseQueryArgs<E = ErrorResponse> = (
 ) & {
   requestHeaders?: RequestHeaders
   prepareHeaders?: PrepareHeaders,
-  customErrors?: (args: ClientError) => E;
+  customErrors?: (args: ClientError) =>  Partial<E | ErrorResponse>;
```

Comment:
Can you explain the reasoning behind this? :)

Reply:
Yes sure, (in the meantime the build has failed with this update :) ). Anyway using just E (that shoud be initialized to ErrorResponse) in VS-Code the autocomplete for the base case doesn't works

in the image below return `foo` should not be possible i'm expecting `{message, stack, name}`

![image](https://user-images.githubusercontent.com/55290106/167819039-be9a5e43-1d34-4886-a3cd-d344fb774361.png)

But if i define the return type it works, (i return `foo` instead of `bar`)

![image](https://user-images.githubusercontent.com/55290106/167819314-aeba2e4d-004b-413c-8a6e-733068c63159.png)

So i tried to add an explicit return type to enforce typings for the base case

Reply:
In the latest commit, i removed the logic discussed before. In my local project it works.
The typization on base case is not so strong (1 image), but if i define a `customError` function i shoud be free to raise the error in the format that i need.
And if i want to enforce the typization i can just pass the type to `graphqlRequestBaseQuery<{myProps: any}>`.
In my opinion it can be fine. Anyway i'm ever open to other suggestions to improve the code ;)